test/libyul/yulOptimizerTests/unusedStoreEliminator/mcopy_overwriting_mcopy.yul

```diff
@@ -0,0 +1,26 @@
+{
+    calldatacopy(0, 0, 0x80)
+
+    let _0 := 0
+    let _64 := 64
+
+    mcopy(_0, _64, _64) // Redundant. MCOPY overwrites it.
+    mcopy(_0, _64, _64)
+
+    return(0, 0x80)
+}
+// ====
+// EVMVersion: >=cancun
+// ----
+// step: unusedStoreEliminator
+//
+// {
+//     {
+//         calldatacopy(0, 0, 0x80)
+//         let _0 := 0
+//         let _64 := 64
+//         mcopy(_0, _64, _64)
```

Comment:
See https://github.com/ethereum/solidity/pull/14779#discussion_r1452498917

Removing `MCOPY` would require adjustments in `UnusedStoreEliminator` to make it properly account for opcodes that perform more than one operation (both read and write). I'm going to try this as a part of changing the codegen and optimizer to make use of the new opcode for 0.8.25, but I'd rather not do that in this PR. It would require even more test coverage and a very careful review.